plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -210,6 +210,11 @@ public void asyncBlobUpload(WriteContext writeContext, ActionListener<Void> comp
         }
     }
 
+    @Override
+    public boolean remoteIntegrityCheckSupported() {
+        return true;
+    }
+
```

Comment:
Curious what does it take for other cloud vendors for implementing remote checksums?

Reply:
Updating checksum on streamed content on backend and then when stream is consumed fully, verifying the resultant checksum of backend with the checksum sent by the client. Checksum updates incur some CPU.

In case of multi-part support by cloud, there is an additional operation of combining all checksums of parts and then doing the verification.

Reply:
Not too worried about the overhead, more concerned about how could we could add a generic support for all cloud vendors. Lets open a separate issue to follow up, since this change is independent of the checksums

Reply:
Created https://github.com/opensearch-project/OpenSearch/issues/9734